docs/contrib/branching.md

```diff
@@ -0,0 +1,87 @@
+---
+title: Branching Model
+---
+
+Software development typically requires a standardized branching model, to
```

Comment:
This document is somehow too general (mentioning 'hidden' topic branches on the developer's machine), also there is some overlap with other documents (`build.md`, `development.md`)